File: src/angular/hq/src/app/models/holiday/get-holiday-v1.ts
import { PagedRequestV1 } from '../common/paged-request-v1';
import { PagedResponseV1 } from '../common/paged-response-v1';
import { SortDirection } from '../common/sort-direction';

export enum Jurisdiciton {
  USA = 1,
  Colombia = 2,
}

export enum SortColumn {
  Name = 1,
  Date = 2,
  Jurisdiciton = 3,
}

export interface GetHolidayV1Request extends PagedRequestV1 {
  search?: string | null;
  id?: string;
  jurisdiciton?: Jurisdiciton;
  sortBy?: SortColumn;
  sortDirection?: SortDirection;
}

export interface GetHolidayV1Record {
  id: string;
  name: string;
  jurisdiciton: Jurisdiciton;
  date?: Date;
}

export interface GetHolidayV1Response
  extends PagedResponseV1<GetHolidayV1Record> {}

File: src/dotnet/HQ.Abstractions/Holiday/GetHolidayV1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using HQ.Abstractions.Common;
using HQ.Abstractions.Enumerations;

namespace HQ.Abstractions.Holiday;

public class GetHolidayV1
{
    public class Request : PagedRequestV1
    {
        public string? Search { get; set; }
        public Guid? Id { get; set; }
        public Jurisdiciton? Jurisdiciton { get; set; }
        public DateOnly? Date { get; set; }
        public bool? UpcomingOnly { get; set; }
        public SortColumn SortBy { get; set; } = SortColumn.Date;
        public SortDirection SortDirection { get; set; } = SortDirection.Asc;
    }

    public enum SortColumn
    {
        Name = 1,
        Date = 2,
        Jurisdiciton = 3
    }

    public class Response : PagedResponseV1<Record>;
    public class Record
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public Jurisdiciton Jurisdiciton { get; set; }
        public DateOnly Date { get; set; }
    }
}